src/script/interpreter.h

```diff
@@ -240,7 +240,7 @@ extern const HashWriter HASHER_TAPLEAF;    //!< Hasher with tag "TapLeaf" pre-fe
 extern const HashWriter HASHER_TAPBRANCH;  //!< Hasher with tag "TapBranch" pre-fed to it.
 
 template <class T>
-uint256 SignatureHash(const CScript& scriptCode, const T& txTo, unsigned int nIn, int nHashType, const CAmount& amount, SigVersion sigversion, const PrecomputedTransactionData* cache = nullptr);
+uint256 SignatureHash(const CScript& scriptCode, const T& txTo, unsigned int nIn, int32_t nHashType, const CAmount& amount, SigVersion sigversion, const PrecomputedTransactionData* cache = nullptr);
```

Comment:
Would it make sense to do this for the remaining `nHashType` types, e.g. in `CTransactionSignatureSerializer`, `ParseSighashString`, `SignTransaction`, `MutableTransactionSignatureCreator`, `SignatureHashOld`, `SignSignature`, etc?

And how can we avoid this error next time, can we enable (a clang-tidy?) check that compares the cpp and header types?

Reply:
> Would it make sense to do this for the remaining nHashType types, e.g. in CTransactionSignatureSerializer, ParseSighashString, SignTransaction, MutableTransactionSignatureCreator, SignatureHashOld, SignSignature, etc?

I don't think it makes sense in this context, but might still be good to do this for consistency. I'd rather change the absolute minimum possible in these files though.

> And how can we avoid this error next time, can we enable (a clang-tidy?) check that compares the cpp and header types?

Maybe we can add a clang-tidy plugin that checks that the names of the type of each of the parameters in the declaration and the definition are the same, but not sure how useful this actually is.  Also a compiler, or rather linker, did actually catch this. Might still be a good exercise to write a plugin for such a check.